python/cudf/dataframe/series.py

```diff
@@ -611,7 +611,12 @@ def dtype(self):
     def _concat(cls, objs, axis=0, index=True):
         # Concatenate index if not provided
         if index is True:
-            index = Index._concat([o.index for o in objs])
+            from cudf.dataframe.multiindex import MultiIndex
+            if isinstance(objs[0].index, MultiIndex):
+                index = MultiIndex._concat([o.index.copy(
+                    deep=False) for o in objs])
```

Comment:
Why do we need shallow copies of the objects feeding into the concat call?

Reply:
The _needs_codes logic modified the .codes dataframe columns, which broke my algorithm by recomputing on intermediate results.

Reply:
Should not be necessary now.